server/graphql/v2/mutation/IndividualMutations.ts

```diff
@@ -0,0 +1,98 @@
+import bcrypt from 'bcrypt';
+import express from 'express';
+import { GraphQLBoolean, GraphQLNonNull, GraphQLString } from 'graphql';
+import { GraphQLDateTime } from 'graphql-scalars';
+
+import activities from '../../../constants/activities';
+import RateLimit, { ONE_HOUR_IN_SECONDS } from '../../../lib/rate-limit';
+import TwoFactorAuthLib from '../../../lib/two-factor-authentication';
+import models from '../../../models';
+import { checkRemoteUserCanUseAccount } from '../../common/scope-check';
+import { RateLimitExceeded, Unauthorized } from '../../errors';
+import { Individual } from '../object/Individual';
+
+const individualMutations = {
+  setChangelogViewDate: {
+    type: new GraphQLNonNull(Individual),
+    description: 'Update the time which the user viewed the changelog updates. Scope: "account".',
+    args: {
+      changelogViewDate: {
+        type: new GraphQLNonNull(GraphQLDateTime),
+      },
+    },
+    resolve: async (_, { changelogViewDate }, req) => {
+      checkRemoteUserCanUseAccount(req);
+
+      const user = await req.remoteUser.update({ changelogViewDate: changelogViewDate });
+      return user.getCollective();
+    },
+  },
+  setNewsletterOptIn: {
+    type: new GraphQLNonNull(Individual),
+    description: 'Update newsletter opt-in preference. Scope: "account".',
+    args: {
+      newsletterOptIn: { type: new GraphQLNonNull(GraphQLBoolean) },
+    },
+    resolve: async (_, { newsletterOptIn }, req) => {
+      checkRemoteUserCanUseAccount(req);
+
+      const user = await req.remoteUser.update({ newsletterOptIn });
+      return user.getCollective();
+    },
+  },
+  setPassword: {
+    type: new GraphQLNonNull(Individual),
+    description: 'Set password to Individual. Scope: "account". 2FA.',
+    args: {
+      password: {
+        type: new GraphQLNonNull(GraphQLString),
+        description: 'The password to set.',
+      },
+      currentPassword: {
+        type: GraphQLString,
+        description: 'The current password (if any) to confirm password change.',
+      },
+    },
+    async resolve(_: void, args, req: express.Request): Promise<Record<string, unknown>> {
+      checkRemoteUserCanUseAccount(req);
+
+      const rateLimit = new RateLimit(`individual_set_password_${req.remoteUser.id}`, 10, ONE_HOUR_IN_SECONDS, true);
+      if (!(await rateLimit.registerCall())) {
+        throw new RateLimitExceeded();
+      }
+
+      // Enforce 2FA
+      const account = req.remoteUser.getCollective();
```

Comment:
Interesting that it doesn't crash `enforceForAccountAdmins`. We should probably add a nil check in the function to prevent that. I also assume that typescript did not detect the error because we haven't typed `Collective` yet.

```suggestion
      const account = await req.remoteUser.getCollective();
```